### PR TITLE
Many doc updates

### PR DESCRIPTION
**Pull request by et**
## Summary

I took a pass at various pages of the docs and cleaned up lots of grammar and inconsistent wording. I did find a bug when linking the ruby docs as well another bug when linking the Go SDK docs.

## How did you test this change?

Visual test

## Are there any deployment considerations?

N/A

### CODE REVIEW

File: highlight.io/components/QuickstartContent/self-host/shared-snippets.tsx [hunk 1]
```diff
@@ -1,4 +1,3 @@
-import { siteUrl } from '../../../utils/urls'
```
Comment (et) [inline highlight.io/components/QuickstartContent/self-host/shared-snippets.tsx:1]: This was an unused import that my editor clean up. Would be nice to add some eslint rules around that like we have in the frontend app:

https://github.com/highlight/highlight/blob/ebb0f1e751caa97b0a3399cfd2fc9615e196e916/frontend/.eslintrc#L46

**Review by Vadman97 — APPROVED** [review]
thank you! all great improvements :)

File: highlight.io/components/QuickstartContent/self-host/shared-snippets.tsx (hunk 1 above)
Comment (Vadman97) [reply highlight.io/components/QuickstartContent/self-host/shared-snippets.tsx:1]: could we share the `.eslintrc` among all packages in the monorepo?

**Review by jay-khatri — APPROVED** [review]

File: highlight.io/components/QuickstartContent/self-host/shared-snippets.tsx (hunk 1 above)
Comment (et) [reply highlight.io/components/QuickstartContent/self-host/shared-snippets.tsx:1]: Probably yes. Would likely require some clean up. I've added that as a hack week idea (#4862).

**Pull request merged**